test/unit-tests.c

```diff
@@ -345,15 +345,26 @@ test_realize_ctx(void **state __attribute__((unused)))
     assert_null(vfu_ctx.pci.caps);
 }
 
+static int
+dummy_attach(vfu_ctx_t *vfu_ctx)
```

Comment:
It's simpler to define this function nested in `test_attach_ctx`, this way you'll have visibility of the actual `vfu_ctx` argument.

Reply:
Done